File: server/internal/application/dto/base_dto.go
package dto

import "time"

// CreateBaseRequest 创建Base请求（对齐原版）
type CreateBaseRequest struct {
	Name    string `json:"name" binding:"required,max=100"`
	Icon    string `json:"icon,omitempty"`
	SpaceID string `json:"spaceId"`  // 从URL路径参数获取，不需要required验证
}

// UpdateBaseRequest 更新Base请求（对齐原版）
type UpdateBaseRequest struct {
	Name string `json:"name,omitempty" binding:"omitempty,max=100"`
	Icon string `json:"icon,omitempty"`
}

// BaseResponse Base响应（对齐原版）
type BaseResponse struct {
	ID        string    `json:"id"`
	Name      string    `json:"name"`
	Icon      string    `json:"icon,omitempty"`
	SpaceID   string    `json:"spaceId"`
	CreatedBy string    `json:"createdBy"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
}

// BaseWithTablesResponse Base带Table列表响应
type BaseWithTablesResponse struct {
	BaseResponse
	Tables []*TableResponse `json:"tables,omitempty"`
}

File: packages/aitable/src/types/infrastructure/base.dto.ts
/**
 * Base DTO - API 层 Base 数据传输对象
 */

/**
 * Base DTO (从 API 返回)
 */
export interface BaseDTO {
  id: string;
  name: string;
  icon?: string;
  spaceId: string;
  createdTime: string;
  lastModifiedTime: string;
}

/**
 * 创建 Base 请求 DTO
 */
export interface CreateBaseDTO {
  name: string;
  icon?: string;
  spaceId: string;
}

/**
 * 更新 Base 请求 DTO
 */
export interface UpdateBaseDTO {
  name?: string;
  icon?: string;
}

